src/server/middleware/artifactinfo/artifact_info_test.go

```diff
@@ -60,8 +60,12 @@ func TestParseURL(t *testing.T) {
 		{
 			input: "/v2/development/golang/manifests/shaxxx:**********************************************************************************************************************************",
 
-			expect: map[string]string{},
-			match:  false,
+			expect: map[string]string{
+				lib.RepositorySubexp: "development/golang",
+				lib.ReferenceSubexp:  "shaxxx:**********************************************************************************************************************************",
+				"tag":                "shaxxx:**********************************************************************************************************************************",
```

Comment:
In this case the tag should be empty.

We should fail early for 400 or 404 at the middleware level.  Otherwise the artifact Info data will be different in the subsequent codepath.